src/DebuggerData.cpp

```diff
@@ -402,20 +379,11 @@ void Breakpoints::loadBreakpoints(QXmlStreamReader& xml)
 		if (xml.isStartElement()) {
 			if (xml.name() == "Breakpoint") {
 				// set type
-				QString type = xml.attributes().value("type").toString().toLower();
-				if (type == "ioread") {
-					bp.type = WATCHPOINT_IOREAD;
-				} else if (type == "iowrite") {
-					bp.type = WATCHPOINT_IOWRITE;
-				} else if (type == "memread") {
-					bp.type = WATCHPOINT_MEMREAD;
-				} else if (type == "memwrite") {
-					bp.type = WATCHPOINT_MEMWRITE;
-				} else if (type == "condition") {
-					bp.type = CONDITION;
-				} else {
-					bp.type = BREAKPOINT;
-				}
+				static QStringList types = (QStringList() << "breakpoint" << "memread" << "memwrite" << "ioread"
+				                                          << "iowrite"    << "condition");
+				QString label = xml.attributes().value("type").toString().toLower();
+				bp.type = static_cast<Breakpoint::Type>(types.indexOf(label));
+				assert(bp.type != -1);
```

Comment:
This assert can trigger when the xml file was edited (so that it contains another value than from this list).
You can argue that the user shouldn't edit this file. But something similar might happen when upgrading from an older to a newer version of the debugger, and the set of types has changed.

Could you share this `types` list between saveBreakpoints() and loadBreakPoints()?
Maybe make this is global constant?  (Global variables should be strongly avoided, but global constants are fine). Either a `static const char* const` or a `static const QStringList` can be made to work, I think.

Reply:
How about this? We add INVALID to type:
```
        enum Type { INVALID = -1, BREAKPOINT, WATCHPOINT_MEMREAD, WATCHPOINT_MEMWRITE,
                    WATCHPOINT_IOREAD, WATCHPOINT_IOWRITE, CONDITION };
```
And in DebuggerData.cpp:
```
           Breakpoint::Type type = static_cast<Breakpoint::Type>(typeStrings.indexOf(label));
           if (type == Breakpoint::INVALID) continue;
           bp.type = type;
```

Reply:
Sure, The important thing is to not trigger an assert (for stuff that can happen in unusual situations).

Personally I wouldn't add an `INVALID = -1` enum value, because that ties the `Type` enum to an implementation detail of how exactly you do the search. E.g. if you would share the array between the load/store function, and search via std::find(), the value `-1` doesn't make sense anymore.

Reply:
Created another PR to solve this one #97